Source/AwsCommonRuntimeKit/io/TlsContextOptions.swift

```diff
@@ -2,14 +2,14 @@
 //  SPDX-License-Identifier: Apache-2.0.
 
 import AwsCIo
-public final class TlsContextOptions {
+public class TlsContextOptions: CStruct {
     private let allocator: Allocator
-    var rawValue: UnsafeMutablePointer<aws_tls_ctx_options>
+    private var rawValue: UnsafeMutablePointer<aws_tls_ctx_options>
     public static func isAlpnSupported() -> Bool {
         return aws_tls_is_alpn_available()
     }
 
-    public init(defaultClientWithAllocator allocator: Allocator = defaultAllocator) {
+    public init(allocator: Allocator = defaultAllocator) {
```

Comment:
the reason this had the weird `defaultClientWithAllocator` name was to make it clear that this was the "default_client" constructor. This class has a LOT of constructors. There's also a "default_server" constructor in C. It's not bound to Swift, but the reasoning behind giving the "default client" function an explicit name was to disambiguate it from other functions we might bind in the future.

It's clunky that we name it "withAllocator" here, because that was the only arg whose name we could customize. I propose that we use factory methods instead of public init functions, so we can give them all good names without abusing the argument names like we are now. Ex:
```
static public makeDefaultClient(allocator: Allocator? = defaultAllocator) -> TlsContextOptions {...}
static public makeClientWithMtls(certificatePath: String, keyPath: String, ...) -> TlsContextOptions {...}
static public makeClientWithMtls(pkcs12Path: String, password: String, ...) -> TlsContextOptions {...}
```

Reply:
Makes sense, thanks.